audit/entry_formatter.go

```diff
@@ -638,3 +647,143 @@ func newTemporaryEntryFormatter(n *EntryFormatter) *EntryFormatter {
 func (s *nonPersistentSalt) Salt(_ context.Context) (*salt.Salt, error) {
 	return salt.NewNonpersistentSalt(), nil
 }
+
+// excludeFields takes an entry (*RequestEntry or *ResponseEntry) and attempts to
+// exclude fields that have been configured on the formatter node.
+// NOTE: Whilst the method accepts 'any' the types must be one of the two specified above.
```

Comment:
I don't think methods (as opposed to funcs) can have generic type constraint parameters 😢 

https://go.googlesource.com/proposal/+/refs/heads/master/design/43651-type-parameters.md#no-parameterized-methods

Reply:
Hmm, damn! We could refactor into a function (that takes `f.exclusions` as an argument, since that is all we use from the EntryFormatter), but ultimately that has larger implications on the code structure than I intended when I suggested the type constraint. I'll leave it up to you -- if you want to toy around with it and go with whatever feels best, I'd be happy either way. It'd be great if we could make it impossible to call the function with anything but an entry, though.

Thanks for humouring me!